includes/Core/Util/User_Input_Settings.php

```diff
@@ -0,0 +1,215 @@
+			$user_input_settings_args['headers']['Content-Type'] = 'application/json';
+			$user_input_settings_args['body']                    = wp_json_encode( $settings );
+		}
+
+		$response = wp_remote_post( $user_input_settings_url, $user_input_settings_args );
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$code = wp_remote_retrieve_response_code( $response );
+		if ( $code < 200 || 299 < $code ) {
+			return new \WP_Error(
+				'user_input_settings_request',
+				__( 'User input settings request failed.', 'google-site-kit' ),
+				array( 'status' => $code )
+			);
+		}
+
+		$body     = wp_remote_retrieve_body( $response );
+		$settings = json_decode( $body, true );
+
+		if ( is_array( $settings ) ) {
+			$this->cache_settings( $settings );
+		}
+
+		return $settings;
+	}
+
+	/**
+	 * Caches user input settings received from the proxy server.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings Array with user input settings.
+	 */
+	private function cache_settings( array $settings ) {
+		$site_settings = array();
+		$user_settings = array();
+
+		foreach ( $settings as $setting_key => $setting_data ) {
+			if ( ! isset( $setting_data['scope'] ) || ! isset( $setting_data['values'] ) ) {
+				continue;
+			}
+
+			$values = is_array( $setting_data['values'] ) ? $setting_data['values'] : array();
+			if ( 'site' === $setting_data['scope'] ) {
+				$site_settings[ $setting_key ] = $values;
+			} elseif ( 'user' === $setting_data['scope'] ) {
+				$user_settings[ $setting_key ] = $values;
+			}
+		}
+
+		$this->transients->set( self::TRANSIENT_NAME, $site_settings, WEEK_IN_SECONDS );
+		$this->user_transients->set( self::TRANSIENT_NAME, $user_settings, WEEK_IN_SECONDS );
+	}
+
+	/**
+	 * Gets user input settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array|WP_Error User input settings.
+	 */
+	public function get_settings() {
+		if ( ! $this->is_connected_to_proxy() ) {
+			return new WP_Error(
+				'not_connected',
+				__( 'Not Connected', 'google-site-kit' ),
+				array( 'status' => 400 )
+			);
+		}
+
+		$data = array(
+			'site' => $this->transients->get( self::TRANSIENT_NAME ),
+			'user' => $this->user_transients->get( self::TRANSIENT_NAME ),
+		);
+
+		if ( ! is_array( $data['site'] ) || ! is_array( $data['user'] ) ) {
+			return $this->sync_with_proxy();
+		}
+
+		$settings = array();
+
+		foreach ( $data as $scope => $values ) {
+			foreach ( $values as $key => $value ) {
+				$settings[ $key ] = array(
+					'values' => $value,
+					'scope'  => $scope,
+				);
+			}
+		}
+
+		return $settings;
+	}
+
+	/**
+	 * Sets user input settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $settings User settings.
+	 * @return array|WP_Error User input settings.
+	 */
+	public function set_settings( $settings ) {
+		return $this->is_connected_to_proxy()
+			? $this->sync_with_proxy( $settings )
+			: new \WP_Error( 'not_connected', __( 'Not Connected', 'google-site-kit' ), array( 'status' => 400 ) );
```

Comment:
Also, please split this onto multiple lines like others.
```suggestion
			: new WP_Error( 'not_connected', __( 'Not Connected', 'google-site-kit' ), array( 'status' => 400 ) );
```